dbms/src/Interpreters/PredicateExpressionsOptimizer.cpp

```diff
@@ -210,7 +228,12 @@ void PredicateExpressionsOptimizer::cloneOuterPredicateForInnerPredicate(
         for (auto projection : projection_columns)
         {
             if (require.second == projection.second)
-                require.first->name = projection.first->getAliasOrColumnName();
+            {
+                ASTPtr & ast = projection.first;
+                if (ast->tryGetAlias().empty())
+                    ast->setAlias(ast->getAliasOrColumnName());
```

Comment:
https://github.com/yandex/ClickHouse/issues/3105#issuecomment-420477177
I tried to solve this problem by set alias.

Reply:
Ok.